test/integration/use_int_test.go

```diff
@@ -139,9 +141,10 @@ func (suite *UseIntegrationTestSuite) TestReset() {
 	cfg, err := config.New()
 	suite.NoError(err)
 	rcfile, err := subshell.New(cfg).RcFile()
-	if runtime.GOOS != "windows" && fileutils.FileExists(rcfile) {
+	rcFilePath := filepath.Join(ts.Dirs.HomeDir, filepath.Base(rcfile))
```

Comment:
Is there some way to have the subshell use the desired path for the rc file rather than reconstructing it after copying contents? I think it's less fragile to make the subshell type configurable and minimize special logic in the tests themselves.